src/zcash/Address.hpp

```diff
@@ -67,6 +67,16 @@ class UnifiedAddress {
 public:
     UnifiedAddress() {}
 
+    static std::optional<UnifiedAddress> ForReceivers(const std::vector<Receiver>& receivers) {
+        UnifiedAddress result;
+        for (const auto& receiver : receivers) {
+            if (!result.AddReceiver(receiver)) {
```

Comment:
This is adding another use of `UnifiedAddress::AddReceiver`, which requires enforcing an assumption about receiver order that is not being enforced here, and which I do not want to risk us accidentally breaking before we fix #5456. Replace this with a `UnifiedAddress::SingleReceiver` method that takes a `Receiver receiver` and returns a `std::optional<UnifiedAddress>`.